.travis.yml

```diff
@@ -5,6 +5,10 @@ services:
   - docker
 language: go
 go_import_path: go.uber.org/yarpc
+branches:
+  only:
+    - master
+    - dev
```

Comment:
Aha, that *is* an unintended consequence indeed. I don't want to affect how we do stacked PRs, even if that buys us some Travis efficiency.

I'm Ok with modifying if @peter-edge thinks that can be accomadated, or reverting if it can't.